src/plugins/vis_type_table/public/to_ast.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getVisSchemas, Vis } from '../../visualizations/public';
+import { buildExpression, buildExpressionFunction } from '../../expressions/public';
+import { TableVisExpressionFunctionDefinition } from './table_vis_fn';
+import { OpenSearchaggsExpressionFunctionDefinition } from '../../data/common/search/expressions';
+
+export const toExpressionAst = (vis: Vis, params: any) => {
+  const opensearchaggs = buildExpressionFunction<OpenSearchaggsExpressionFunctionDefinition>(
+    'opensearchaggs',
+    {
+      index: vis.data.indexPattern!.id!,
+      metricsAtAllLevels: vis.isHierarchical(),
+      partialRows: vis.params.showPartialRows || false,
+      aggConfigs: JSON.stringify(vis.data.aggs!.aggs),
+      includeFormatHints: false,
+    }
+  );
+
+  const schemas = getVisSchemas(vis, params);
+  // remove duplicate metrics
+  const metrics =
+    schemas.bucket && vis.params.showPartialRows && !vis.params.showMetricsAtAllLevels
+      ? schemas.metric.slice(-1 * (schemas.metric.length / schemas.bucket.length))
```

Comment:
this `.slice` could benefit from an explanatory statement.

Reply:
will add some comments here

Reply:
Ideally we'd also have a test case in the unit test file (now that it exists) to validate this behavior 😁